### PR TITLE
[fix][readme]update deploy-kubernetes url

### LINKED ISSUE #21195: [Doc] Kubernetes installation doc link broken
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### What issue do you find in Pulsar docs?

The current link available [here](https://github.com/apache/pulsar/tree/master/deployment/kubernetes) to Kubernetes installation is forwarding to a broken link https://github.com/apache/pulsar/blob/master/site2/docs/deploy-kubernetes.md

### What is your suggestion?

Fix the link to the official Kubernetes installation guide

### Any reference?

This seems to be the correct link for this case: https://pulsar.apache.org/docs/3.1.x/deploy-kubernetes/

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by sunheyi6**
Fixes #21195 

Main Issue: #xyz

PIP: #xyz 

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If the box was checked, please highlight the changes*

- [ ] Dependencies (add or upgrade a dependency)
- [ ] The public API
- [ ] The schema
- [ ] The default values of configurations
- [ ] The threading model
- [ ] The binary protocol
- [ ] The REST endpoints
- [ ] The admin CLI options
- [ ] The metrics
- [ ] Anything that affects deployment

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete` 

### Matching PR in forked repository

PR in forked repository:

### CODE REVIEW

**Review by lhotari — COMMENTED** [review]
Please link to https://pulsar.apache.org/docs/next/deploy-kubernetes/ so that it stays current.

**Comment by sunheyi6** [comment]
> Please link to https://pulsar.apache.org/docs/next/deploy-kubernetes/ so that it stays current.请链接到 https://pulsar.apache.org/docs/next/deploy-kubernetes/ 以使其保持最新状态。

already updated

File: deployment/kubernetes/README.md [hunk 1]
```diff
@@ -22,5 +22,5 @@
 This directory contains the Kubernetes services definitions for all
 the components required to do a complete Pulsar deployment.
 
-Refer to [Kubernetes.md](../../site2/docs/deploy-kubernetes.md) document for instructions on
+Refer to [Kubernetes.md](https://pulsar.apache.org/docs/next/deploy-kubernetes/) document for instructions on
```
Comment (tisonkun) [inline deployment/kubernetes/README.md:25]:
```suggestion
Refer to [Kubernetes.md](https://pulsar.apache.org/docs/deploy-kubernetes/) document for instructions on
```

**Review by tisonkun — APPROVED** [review]

**Comment by tisonkun** [comment]
Thank you! Merging...

**Pull request merged**